app/controllers/admins/sessions_controller.rb

```diff
@@ -1,15 +1,25 @@
 class Admins::SessionsController < Devise::SessionsController
- before_filter :check_subdomain?, only: [:new]
+  before_action :check_subdomain?, only: [:new]
+  before_action :check_admin_company_and_status, only: [:create]
 
   protected
 
   def check_subdomain?
     subdomain = request.subdomain
-    if ['www', 'ftp', 'ssh'].include?(subdomain)
-      return redirect_to root_url(host: Rails.application.config.action_mailer.default_url_options[:host])
+    if subdomain == 'www'
+      redirect_to root_url(host: Rails.application.config.action_mailer.default_url_options[:host])
     elsif subdomain.blank? || Company.find_by(subdomain: subdomain).nil?
-      return redirect_to new_company_url(host: Rails.application.config.action_mailer.default_url_options[:host]), alert: 'Get Registered First'
+      flash[:alert] = 'Get Registered First'
+      redirect_to root_url(host: Rails.application.config.action_mailer.default_url_options[:host])
     end
   end
 
+  def check_admin_company_and_status
+    admin = Admin.where(email: params[:admin][:email]).first
+    if admin.nil? || admin.company.subdomain != request.subdomain
+      return redirect_to sign_in_path, alert: 'You are not authorized'
```

Comment:
if i don't use return then it will go to the other return also and give us double render error.